gusto/timeloop.py

```diff
@@ -42,7 +32,8 @@ def _set_ubar(self):
         unp1 = state.xnp1.split()[0]
 
         for field, advection in self.advection_dict.iteritems():
-            advection.ubar.assign(un + state.timestepping.alpha*(unp1-un))
+            if not isinstance(advection, NoAdvection):
+                advection.ubar.assign(un + state.timestepping.alpha*(unp1-un))
```

Comment:
This kind of instance check is bad code smell.  The advection object should have a method to update `ubar` given some expression for the new `ubar`, and then `NoAdvection` can just implement a no-op there.  Then the caller doesn't need to know about what the object type is.

i.e., something like:

```python
class Advection(object):
    ...
    def update_ubar(self, expr):
        self.ubar.assign(expr)

class NoAdvection(Advection):
    def update_ubar(self, expr):
        return
```